src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int amountInStore = FruitStorage.fruitRepository.get(fruit);
+        int amountToBuy = fruitTransaction.getQuantity();
+        if (amountInStore < amountToBuy) {
+            throw new RuntimeException("Not enough fruit in the storage for this operation");
+        }
+        FruitStorage.fruitRepository.put(fruit, amountInStore - amountToBuy);
+    }

```

Comment:
The current implementation of the `PurchaseOperation` class could lead to negative values in the storage, which violates the checklist requirement: 'Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.'
To adhere to this requirement, you should add an additional check after line 15 to ensure that the resulting amount in storage is not negative before updating the storage.